Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1925,27 +1931,28 @@ private static ErrorInformation CreateErrorInformation(Counterexample error, VC.
         }
         else
         {
-          var msg = assertError.FailingAssert.ErrorData as string;
-          var tok = assertError.FailingAssert.tok;
-          if (!CommandLineOptions.Clo.ForceBplErrors && assertError.FailingAssert.ErrorMessage != null)
+          string msg = null;
+          string bec = null;
+          if (assertError.FailingAssert.ErrorMessage == null || CommandLineOptions.Clo.ForceBplErrors)
+          {
+            msg = assertError.FailingAssert.ErrorData as string;
+            if (msg == null)
+            {
+              msg = "This assertion might not hold.";
+              bec = "BP5001";
```

Comment:
Should we get rid of this case distinction and set `bec` in both cases? This would be consistent to pre/postcondition violations.